perllib/FixMyStreet/Queue/Item/Report.pm

```diff
@@ -202,27 +219,78 @@ sub _create_reporters {
 
     if (FixMyStreet->staging_flag('send_reports', 0)) {
         # on a staging server send emails to ourselves rather than the bodies
-        %reporters = map { $_ => $reporters{$_} } grep { /FixMyStreet::SendReport::Email/ } keys %reporters;
-        unless (%reporters) {
-            %reporters = ( 'FixMyStreet::SendReport::Email' => FixMyStreet::SendReport::Email->new() );
+        unless (%email_reporters) {
+            %email_reporters
+                = ( EMAIL_SENDER_PREFIX => ${ \EMAIL_SENDER_PREFIX }->new() );
         }
+
+        @other_reporters = ();
     }
 
-    $self->_set_reporters(\%reporters);
+    $self->_set_reporters([@other_reporters, values %email_reporters]);
 }
 
 sub _send {
     my $self = shift;
 
     # Multiply results together, so one success counts as a success.
     my $result = -1;
+    my $report = $self->report;
+
+    my @add_send_fail_body_ids;
+    my @remove_send_fail_body_ids;
+
+    for my $sender ( @{ $self->reporters } ) {
+        my $sender_name = ref $sender;
+        $self->log( 'Sending using ' . $sender_name );
+
+        if ( FixMyStreet->staging_flag( 'send_reports', 0 ) ) {
+            # If on staging with send_reports set to 0, bypass the body_id
+            # logic below, as we may only have a special Email reporter with
+            # no bodies attached
+            my $res = $sender->send( $report, $self->h );
+            $result *= $res;
+            $report->add_send_method($sender_name) if !$res;
+        }
+        elsif ( $sender_name =~ /${\EMAIL_SENDER_PREFIX}/ ) {
+            # Email is special-cased; an email to e.g. a two-tier council
+            # will have multiple recipients in the 'To' line. So we only
+            # need to record one failure (otherwise on retry/success,
+            # duplicate emails will be sent).
+            my $res = $sender->send( $self->report, $self->h );
+
+            $result *= $res;
+
+            my @body_ids = map { $_->id } @{ $sender->bodies };
+            if ($res) {
+                push @add_send_fail_body_ids, @body_ids;
+            }
+            else {
+                $report->add_send_method($sender_name);
+                push @remove_send_fail_body_ids, @body_ids;
+            }
+        }
```

Comment:
Don't generally mind, but could the `else`s etc be on the same line as the braces, as per our coding standards (and the rest of this file), thanks.

Reply:
I've tried to fix this but I may have missed some.